### ISSUE TITLE
Simulation.get_array_metadata throws error when called with 1D simulation with dimension=3

### ISSUE THREAD

**Issue by edmundsj** [NanoComp/meep#1235]
I am running 1D simulations with dimension=3 (due to the desire to run them at oblique incidence with Bloch periodic boundary conditions), and when I try to call get_array_metadata MEEP throws an error. I do not have the same problem when I force the simulation to be 1D using dimensions=1 as a simulation argument.

```
import meep as mp
import numpy as np
import matplotlib.pyplot as plt
from matplotlib import animation

resolution = 32
frequency = 2.0
frequencyWidth = 1
numberFrequencies = 1
pmlThickness = 2.0
animationTimestepDuration = 0.05
materialThickness = 2.5
length = 2 * materialThickness + 2 * pmlThickness
powerDecayTarget = 1e-9

transmissionMonitorLocation = mp.Vector3(0, 0, materialThickness/4)

cellSize = mp.Vector3(0, 0, length)
sources = [mp.Source(
    mp.GaussianSource(frequency=frequency,fwidth=frequencyWidth),
    component=mp.Ex,
    center=mp.Vector3(0, 0, -materialThickness/2)),
    ]

geometry = [mp.Block(
    mp.Vector3(mp.inf, mp.inf, materialThickness + pmlThickness),
    center=mp.Vector3(0, 0, materialThickness/2 + pmlThickness/2),
    material=mp.Medium(index=2))]

pmlLayers = [mp.PML(thickness=pmlThickness, R_asymptotic=1e-35)]
absorberLayers = [mp.Absorber(thickness=2.0)]

simulation = mp.Simulation(
    cell_size=cellSize,
    sources=sources,
    resolution=resolution,
    boundary_layers=pmlLayers)

incidentRegion = mp.FluxRegion(center=mp.Vector3(0, 0, -materialThickness/4))
incidentFluxMonitor = simulation.add_flux(frequency, frequencyWidth, numberFrequencies, incidentRegion)

simulation.run(until_after_sources=mp.stop_when_fields_decayed(20, mp.Ex,
    transmissionMonitorLocation, powerDecayTarget))
fieldEx = np.real(simulation.get_array(center=mp.Vector3(0, 0, 0), size=cellSize, component=mp.Ex))
fieldData = np.zeros(len(fieldEx))

incidentFluxToSubtract = simulation.get_flux_data(incidentFluxMonitor)

simulation = mp.Simulation(
    cell_size=cellSize,
    sources=sources,
    resolution=resolution,
    boundary_layers=pmlLayers,
    geometry=geometry)

transmissionRegion = mp.FluxRegion(center=transmissionMonitorLocation)
transmissionFluxMonitor = simulation.add_flux(frequency, frequencyWidth, numberFrequencies, transmissionRegion)
reflectionRegion = incidentRegion
reflectionFluxMonitor = simulation.add_flux(frequency, frequencyWidth, numberFrequencies, reflectionRegion)
simulation.load_minus_flux_data(reflectionFluxMonitor, incidentFluxToSubtract)

def updateField(sim):
    global fieldData
    fieldEx = np.real(sim.get_array(center=mp.Vector3(0, 0, 0), size=cellSize, component=mp.Ex))
    fieldData = np.vstack((fieldData, fieldEx))

simulation.run(mp.at_every(animationTimestepDuration, updateField),
        until_after_sources=mp.stop_when_fields_decayed(20, mp.Ex,
            transmissionMonitorLocation, powerDecayTarget))

incidentFlux = np.array(mp.get_fluxes(incidentFluxMonitor))
transmittedFlux = np.array(mp.get_fluxes(transmissionFluxMonitor))
reflectedFlux = np.array(mp.get_fluxes(reflectionFluxMonitor))
R = -reflectedFlux / incidentFlux
T = transmittedFlux / incidentFlux

simulation.get_array_metadata()
```
Output of MEEP:

```
(mp) C02X32GFJGH5:examples jordan.e$ python bragg_mirror.py
-----------
Initializing structure...
time for choose_chunkdivision = 9.01222e-05 s
Working in 3D dimensions.
Computational cell is 0.03125 x 0.03125 x 9 with resolution 32
time for set_epsilon = 0.00209093 s
-----------
field decay(t = 20.015625): 0.2591139729589917 / 0.2591139729589917 = 1.0
field decay(t = 40.03125): 4.791648710632854e-15 / 0.2591139729589917 = 1.8492436574970805e-14
run 0 finished at t = 40.03125 (2562 timesteps)
-----------
Initializing structure...
time for choose_chunkdivision = 6.00815e-05 s
Working in 3D dimensions.
Computational cell is 0.03125 x 0.03125 x 9 with resolution 32
     block, center = (0,0,2.25)
          size (1e+20,1e+20,4.5)
          axes (1,0,0), (0,1,0), (0,0,1)
          dielectric constant epsilon diagonal = (4,4,4)
time for set_epsilon = 0.00267482 s
-----------
field decay(t = 20.015625): 0.12470870574723257 / 0.12470870574723257 = 1.0
field decay(t = 40.03125): 6.036898285939192e-14 / 0.12470870574723257 = 4.840799405115434e-13
run 0 finished at t = 40.03125 (2562 timesteps)
[0.9071835]
[0.0928165]
[1.]
Traceback (most recent call last):
  File "bragg_mirror.py", line 99, in <module>
    simulation.get_array_metadata()
  File "/Users/jordan.e/opt/miniconda3/envs/mp/lib/python3.7/site-packages/meep/simulation.py", line 1969, in get_array_metadata
    xyzw_vector = self.fields.get_array_metadata(v, collapse, snap)
  File "/Users/jordan.e/opt/miniconda3/envs/mp/lib/python3.7/site-packages/meep/__init__.py", line 3906, in get_array_metadata
    return _meep.fields_get_array_metadata(self, where, collapse_empty_dimensions, snap_empty_dimensions)
RuntimeError: meep: weird yee shift for component none
```

**Comment by oskooi** [comment]
Calling `get_array_metadata` works fine for me:
```
>>> exec(open("get_array_metadata_bug.py").read())
Using MPI version 3.1, 1 processes
-----------
Initializing structure...
time for choose_chunkdivision = 4.8017e-05 s
Working in 3D dimensions.
Computational cell is 0.03125 x 0.03125 x 9 with resolution 32
time for set_epsilon = 0.00192776 s
-----------
field decay(t = 20.015625): 0.25911397295899147 / 0.25911397295899147 = 1.0
field decay(t = 40.03125): 4.791648711325046e-15 / 0.25911397295899147 = 1.84924365776422e-14
run 0 finished at t = 40.03125 (2562 timesteps)
-----------
Initializing structure...
time for choose_chunkdivision = 4.1341e-05 s
Working in 3D dimensions.
Computational cell is 0.03125 x 0.03125 x 9 with resolution 32
     block, center = (0,0,2.25)
          size (1e+20,1e+20,4.5)
          axes (1,0,0), (0,1,0), (0,0,1)
          dielectric constant epsilon diagonal = (4,4,4)
time for set_epsilon = 0.00306151 s
-----------
field decay(t = 20.015625): 0.12470870574723261 / 0.12470870574723261 = 1.0
field decay(t = 40.03125): 6.03689828443433e-14 / 0.12470870574723261 = 4.840799403908731e-13
run 0 finished at t = 40.03125 (2562 timesteps)
>>> simulation.get_array_metadata()
[(-0.046875, -0.015625), (-0.046875, -0.015625), (-4.484375, -4.453125, -4.421875, -4.390625, -4.359375, -4.328125, -4.296875, -4.265625, -4.234375, -4.203125, -4.171875, -4.140625, -4.109375, -4.078125, -4.046875, -4.015625, -3.984375, -3.953125, -3.921875, -3.890625, -3.859375, -3.828125, -3.796875, -3.765625, -3.734375, -3.703125, -3.671875, -3.640625, -3.609375, -3.578125, -3.546875, -3.515625, -3.484375, -3.453125, -3.421875, -3.390625, -3.359375, -3.328125, -3.296875, -3.265625, -3.234375, -3.203125, -3.171875, -3.140625, -3.109375, -3.078125, -3.046875, -3.015625, -2.984375, -2.953125, -2.921875, -2.890625, -2.859375, -2.828125, -2.796875, -2.765625, -2.734375, -2.703125, -2.671875, -2.640625, -2.609375, -2.578125, -2.546875, -2.515625, -2.484375, -2.453125, -2.421875, -2.390625, -2.359375, -2.328125, -2.296875, -2.265625, -2.234375, -2.203125, -2.171875, -2.140625, -2.109375, -2.078125, -2.046875, -2.015625, -1.984375, -1.953125, -1.921875, -1.890625, -1.859375, -1.828125, -1.796875, -1.765625, -1.734375, -1.703125, -1.671875, -1.640625, -1.609375, -1.578125, -1.546875, -1.515625, -1.484375, -1.453125, -1.421875, -1.390625, -1.359375, -1.328125, -1.296875, -1.265625, -1.234375, -1.203125, -1.171875, -1.140625, -1.109375, -1.078125, -1.046875, -1.015625, -0.984375, -0.953125, -0.921875, -0.890625, -0.859375, -0.828125, -0.796875, -0.765625, -0.734375, -0.703125, -0.671875, -0.640625, -0.609375, -0.578125, -0.546875, -0.515625, -0.484375, -0.453125, -0.421875, -0.390625, -0.359375, -0.328125, -0.296875, -0.265625, -0.234375, -0.203125, -0.171875, -0.140625, -0.109375, -0.078125, -0.046875, -0.015625, 0.015625, 0.046875, 0.078125, 0.109375, 0.140625, 0.171875, 0.203125, 0.234375, 0.265625, 0.296875, 0.328125, 0.359375, 0.390625, 0.421875, 0.453125, 0.484375, 0.515625, 0.546875, 0.578125, 0.609375, 0.640625, 0.671875, 0.703125, 0.734375, 0.765625, 0.796875, 0.828125, 0.859375, 0.890625, 0.921875, 0.953125, 0.984375, 1.015625, 1.046875, 1.078125, 1.109375, 1.140625, 1.171875, 1.203125, 1.234375, 1.265625, 1.296875, 1.328125, 1.359375, 1.390625, 1.421875, 1.453125, 1.484375, 1.515625, 1.546875, 1.578125, 1.609375, 1.640625, 1.671875, 1.703125, 1.734375, 1.765625, 1.796875, 1.828125, 1.859375, 1.890625, 1.921875, 1.953125, 1.984375, 2.015625, 2.046875, 2.078125, 2.109375, 2.140625, 2.171875, 2.203125, 2.234375, 2.265625, 2.296875, 2.328125, 2.359375, 2.390625, 2.421875, 2.453125, 2.484375, 2.515625, 2.546875, 2.578125, 2.609375, 2.640625, 2.671875, 2.703125, 2.734375, 2.765625, 2.796875, 2.828125, 2.859375, 2.890625, 2.921875, 2.953125, 2.984375, 3.015625, 3.046875, 3.078125, 3.109375, 3.140625, 3.171875, 3.203125, 3.234375, 3.265625, 3.296875, 3.328125, 3.359375, 3.390625, 3.421875, 3.453125, 3.484375, 3.515625, 3.546875, 3.578125, 3.609375, 3.640625, 3.671875, 3.703125, 3.734375, 3.765625, 3.796875, 3.828125, 3.859375, 3.890625, 3.921875, 3.953125, 3.984375, 4.015625, 4.046875, 4.078125, 4.109375, 4.140625, 4.171875, 4.203125, 4.234375, 4.265625, 4.296875, 4.328125, 4.359375, 4.390625, 4.421875, 4.453125, 4.484375), array([[[ 2.73437500e-002,  3.12500000e-002,  3.12500000e-002, ...,
          3.12500000e-002,  2.73437500e-002,  3.90625000e-003],
        [ 1.14623230e-320,  5.58294180e-322,  9.88131292e-324, ...,
          0.00000000e+000,  4.50000000e+000,  0.00000000e+000]],

       [[ 0.00000000e+000,  2.42092166e-322,  0.00000000e+000, ...,
          0.00000000e+000,  0.00000000e+000,  6.89882052e-310],
        [ 1.00000000e+000,  0.00000000e+000,  0.00000000e+000, ...,
          6.89897908e-310, -4.50000000e+000,  1.63041663e-322]]])]
```
Perhaps the Conda package you are using is out of date and contains an old bug for `get_array_metadata`? You can find out the version you are using with: `print(mp.__version__)`. Try updating to the most recent version (1.14).

**Comment by edmundsj** [comment]
I'm running OS X, looks like the latest version available through conda is 1.12, which I am updated to. I suspect you are correct and this was fixed by 1.14. Are their plans to update the mac version on conda?

**Comment by oskooi** [comment]
Unfortunately, we've run into some issues upgrading the Conda package for the [stable/official release](https://meep.readthedocs.io/en/latest/Installation/#official-releases) for OSX which, as you discovered, is currently 1.12 (the Linux version though is up to date). There are two possible workarounds: (1) use the Conda package for the [nightly build](https://meep.readthedocs.io/en/latest/Installation/#nightly-builds) which is up to date for OSX or (2) [build from source](https://meep.readthedocs.io/en/latest/Build_From_Source/).
